noir/noir-repo/compiler/noirc_frontend/src/hir/def_collector/dc_mod.rs

```diff
@@ -41,16 +42,28 @@ pub fn collect_defs(
     module_id: LocalModuleId,
     crate_id: CrateId,
     context: &mut Context,
+    macro_processors: &Vec<&dyn MacroProcessor>,
```

Comment:
This complicates things a lot, but will try to move it to the next processor (`process_unresolved_trait_impls`). Even if we're gonna ditch this macro system eventually, I still think having access to the whole AST in the first pass would be valuable.

Reply:
`process_unresolved_trait_impls` was made very specific as to avoid passing the full def collector but this could be generalized to `process_collected_defs` (as you mentioned in the other comment) which does access the whole AST. We can then ditch `process_untyped_ast` and only have macro processors used in `dc_crate.rs`

Reply:
Closing this discussion after friday's meeting and addressing the macro concerns [here](https://github.com/AztecProtocol/aztec-packages/pull/5127/files)